lock-api/src/test/java/com/palantir/lock/client/CoalescingTransactionServiceTest.java

```diff
@@ -0,0 +1,245 @@
+        verify(lockLeaseService).startTransactions(3);
+    }
+
+    private void assertThatStartTransactionResponsesAreUnique(StartIdentifiedAtlasDbTransactionResponse... responses) {
+        assertThatStartTransactionResponsesAreUnique(Arrays.asList(responses));
+    }
+
+    private void assertThatStartTransactionResponsesAreUnique(
+            List<StartIdentifiedAtlasDbTransactionResponse> responses) {
+        assertThat(responses)
+                .as("Each response should have a different immutable ts lock token")
+                .extracting(response -> response.immutableTimestamp().getLock().getRequestId())
+                .doesNotHaveDuplicates();
+
+        assertThat(responses)
+                .as("Each response should have a different start timestamp")
+                .extracting(response -> response.startTimestampAndPartition().timestamp())
+                .doesNotHaveDuplicates();
+    }
+
+    private void assertDerivableFromBatchedResponse(
+            StartIdentifiedAtlasDbTransactionResponse startTransactionResponse,
+            StartTransactionResponseV4 batchedStartTransactionResponse) {
+
+        assertThat(startTransactionResponse.immutableTimestamp().getLock())
+                .as("Should have a lock token share referencing to immutable ts lock token")
+                .isInstanceOf(LockTokenShare.class)
+                .extracting(t -> ((LockTokenShare) t).sharedLockToken())
+                .isEqualTo(batchedStartTransactionResponse.immutableTimestamp().getLock());
+
+        assertThat(startTransactionResponse.immutableTimestamp().getImmutableTimestamp())
+                .as("Should have same immutable timestamp")
+                .isEqualTo(batchedStartTransactionResponse.immutableTimestamp().getImmutableTimestamp());
+
+        assertThat(startTransactionResponse.startTimestampAndPartition().partition())
+                .as("Should have same partition value")
+                .isEqualTo(batchedStartTransactionResponse.timestamps().partition());
+
+        assertThat(batchedStartTransactionResponse.timestamps().stream())
+                .as("Start timestamp should be contained by batched response")
+                .contains(startTransactionResponse.startTimestampAndPartition().timestamp());
+    }
+
+
+    private static StartTransactionResponseV4 getStartTransactionResponse(
+            long lowestStartTs, int count) {
+        return StartTransactionResponseV4.of(
+                IMMUTABLE_TS_RESPONSE,
+                getPartitionedTimestamps(lowestStartTs, count),
+                LEASE);
+    }
+
+    private static PartitionedTimestamps getPartitionedTimestamps(long startTs, int count) {
+        return ImmutablePartitionedTimestamps.builder()
+                .start(startTs)
+                .count(count)
+                .interval(NUM_PARTITIONS)
+                .build();
+    }
+
+    private static final class BlockingBatchedResponse implements Answer<StartTransactionResponseV4> {
```

Comment:
We have tests for autobatcher, I'm not sure it's useful to prove that it works everywhere we use it imo.